zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+======= ================== ========================== =========================================================================
+Varies  nIssueActions      compactSize                The number of issuance actions in the bundle
+Varies  vIssueActions      IssueAction[nIssueActions] A sequence of issuance actions descriptions
+32      ik                 byte[32]                   The public verification key of the issuer, used to validate the signature
+64      authorization      byte[64]                   The signature of the transaction SIGHASH, signed by the issuer
+======= ================== ========================== =========================================================================
+
+Issuance Protocol
+-----------------
+The issuer program performs the following operations
+
+For all actions `IssueAction`:
+- encode `asset_desc` as 512 byte-string
+- compute `AssetId` as :math:`GroupHash^{\mathbb{P}}(ik || asset_desc)`
+- set the `finalize` boolean as desired (if more more issuance actions are to be created for this asset identifier, set `finalize = 0`, otherwise set `finalize = 1`)
+- For each recipient `i`:
+    - generate a ZSA output note as :math:`\mathsf{ note_i =(d_i, {pk_d}_i, v_i, \rho_i, \psi_i, \mathsf{AssetId}, rcm_i)}`
+- encode the `IssueAction` into the vector `vIssueActions` of the bundle
+
+For the `IssueBundle`:
+- encode the `vIssueActions` vector
+- encode the `ik` as 32 byte-string
+- sign the `SIGHASH` of the transaction with the `issuanceAuthorizingKey`, `isk`, using the RedPallas signature scheme. The signature is then added to the issuance bundle.
+
+
+NOTE that the commitment is not included in the `IssuanceAction` itsefl. As explained below, it is later computed by the validators and added to the `NoteCommitmentTree`.
+
+Concrete applications
+---------------------
+
+**Bridging Assets**
+Issuers can wrap assets defined in other chains and issue them at once in a single transaction. The specifics of the bridge are not implemented in the protocol, but there are several ways we conceive issuers can build these bridges (at least centralized bridges):
+- First, the issuance and burn mechanism can be used in conjunction to determine the 
+
+**Asset Features**
+- By using the `finalize` boolean and the burning mechanism defined in [#zip-0226]_, issuers can control the supply production of any asset associated to their issuer keys. For example,
+    - by setting `finalize = 1` from the first issuance action for that asset type, the issuer is in essence creating a one-time issuance transaction. This is useful when the max supply is capped from the beginning and the distribution is known in advance. All tokens are issued at once and distributed as needed.
+- Issuers can also stop the existing supply production of any asset associated to their issuer keys. This could be done by
+    - issuing a last set of tokens of that specific `AssetId`, while at the same time setting `finalize = 1`, or by
```

Comment:
```suggestion
    - issuing a last set of tokens of that specific `AssetId` during which `finalize = 1`, or by
```

Reply:
This is executing an `Issuance Action`, which hasn't really been defined by this point in the document

Reply:
Done.

Reply:
Have added a bullet clarifying this.